src/main/scala/DateParserApp.scala

```diff
@@ -0,0 +1,57 @@
+import wvlet.log.{LogLevel, LogSupport, Logger}
+
+import java.text.{ParseException, SimpleDateFormat}
+import java.util.Date
+import scala.util.Try
+
+
+/** Collection of utils to parse the date.
+ *
+ * @param log_level set logger level
+ * */
+class DateParserApp(log_level: LogLevel = LogLevel.INFO) extends LogSupport {
+
+  // Prepare logger
+  Logger.init
+  Logger.setDefaultLogLevel(log_level)
+  info(s"Logger set to ${Logger.getDefaultLogLevel} level")
```

Comment:
1. We use `slf4j` api for logging. It is up to the runner to attach to the proper implementation.
2. Log levels should be set in config (log4j.properties, since Spark uses Log4J implementation), remove the log level from the constructor

The logger is almost always (with some very special exceptions to the rule) initialized as follows:
```scala
import org.slf4j.LoggerFactory

class SomeClass {
  private val log = LoggerFactory.getLogger(this.getClass)

  // The rest of the class implementation
}
```

Reply:
took this as nit